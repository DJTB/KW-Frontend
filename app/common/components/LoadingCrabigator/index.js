import React from 'react';
import styled, { keyframes } from 'styled-components';

import { transparent } from 'common/styles/colors';
import { gutter } from 'common/styles/layout';

const fadeOut = keyframes`
  50% { opacity: 0; }
`;
const fadeIn = keyframes`
  50% { opacity: 1; }
`;

const LayoutWrapper = styled.div`
  ${gutter({ type: 'outer', mod: 2 })}
  display: flex;
  width: 100%;
  height: 100%;
  flex: 1 0 100%;
  justify-content: center;
  align-items: center;
  align-content: center;
`;

const SvgWrapper = styled.div`
  position: relative;
  width: 30vmin;
  height: 20vmin;
  background-repeat: no-repeat;
  background-color: ${transparent};
  transition: all 200ms ease-in-out;
`;

const WalkingCrabigator = styled.svg`
  display: block;
  pointer-events: none;
  transform-origin: 50% 50% 0px;
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;

  /* dance crabby dance */
  & path[id*="Step1"] {
    animation: ${fadeOut} 1s steps(1) infinite;
  }
  & path[id*="Step2"] {
    animation: ${fadeIn} 1s steps(1) infinite;
  }
`;

const LoadingCrabigator = () => (
  <LayoutWrapper>
    <SvgWrapper>
      <WalkingCrabigator
        title="Loading Animation"
        width="100%"
        height="100%"
        viewBox="0 0 400 200"
        preserveAspectRatio="xMinYMin meet"
      >
        <path id="eyeBg" fill="#fff" d="M84 30h20v19H84z" />
        <path id="pinkClawStep1" fill="#f36353" d="M135 155c1.3 1.7 0 6.3 0 6.3h-6.3v6.2h-12.4v6.3h-12.5v6.2H85v-6.3H72.5v-12.4h6.3V155H85v-6.3h6.3v-6.2S126 143 135 155zm-43.8 15.6v-3H85v6.2h6.3zm0-28h6.3z" />
        <path id="neckBg" fill="#d0efdb" d="M79 92h44v38H79z" />
        <path id="pinkClawStep2" opacity="0" fill="#f36353" d="M103.8 167.5H91.3v6.3H78.8v-12.6H85V155h6.3v-17l12.8.8z" />
        <path id="shell" fill="#ba8a59" d="M228.8 167.5h-12.6v-6.3H160V155h-13c.4-45.7 1-49 .5-75H160v-6.3h25v-6.2h50v6.3h18.8V80H266v6.3h6.3v6.2h6.3v6.3h6.2v12.5h6.3v12.5h6.6s-.4 5.2 0 12.5c1.6 27.4 4.5 24.5-18.8 25-20.7.3-12-3.8-12.4 6.2h-25z" />
        <path id="headTail" fill="#8ac249" d="M122.5 92.5h-6.3v6.3H78.8v-6.3h-6.2v-6.3H60V80h-6.3v-6.3H41.3v-6.2H35v-6.3h-6.3V55h-6.2V42.5h6.3v-6.3h18.7v6.3h6.3v-6.3h12.5V30h12.5v-6.3h18.7V30H110v6.3h12.5v6.2h6.3v6.3h6.2V55h6.3v12.5h6.2V80h6.3c0 20-8.3 30-11.3 44-3.8 17-20.5 17.7-20.5 5.6l.5-37zm-18.8-50v-6.3h-6.2V30h-6.3v6.3H85v6.2h6.3v6.3h12.5zm-12.5-3V36h6.3v6.3h-6.3zM285 155v-6.3h6.3v-12.4H310v6.2h43.8v-6.3H366V130h6.3v12.5H366v6.3h-6v6.2h-18.8v6.3H285z" />
        <path id="clawLegsStep1" fill="#d45720" d="M222.5 183v-9h-6.3v-12.6h-18.7v6.3h-6.3v6.3h-12.4v6.2h-6.3v6.3h-6.3v-18.8h6.3v-6.3h6.3V130h18.7v6.3h6.3v6.2h18.7v6.3H235v12.4h6.3V155h6.2v-6.3h25v18.8h6.3V180h-6.3v6.3h-6.3v-12.6H260v-6.2h-25v18.8h-6.3v6.2h-6.2zM85 177v-3.2h6.3v-12.6h6.2v-25h-6.3v6.3H78.8v6.3H66.3v6.2h-25v-6.3H35V130h6.3v-6.3h6.2v-6.2h6.3v-6.3h12.5V105H85v6.3h6.3v6.2h6.2v6.3h6.3v6.2h25v-6.3h12.4v-6.2h25v18.8H160v6.2h-6.3v6.3h-6.2v6.2H135v6.3h-6.3v6.2h-12.4v6.3h-12.5v6.2H85zm-37.5-31.4v-3h-6.3v6.2h6.3z" />
        <path id="clawLegsStep2" opacity="0" fill="#d45720" d="M241.3 183v-3h-12.6v-12.5h-12.4v-6.3h-25v6.3H185v6.3h-12.5v6.2h-18.8v-6.3h6.3v-6.2h6.3v-6.3h6.2v-18.7h6.3V130h18.7v6.3h6.3v6.2h18.7v6.3H235v12.4h6.3V155h6.2v-6.3h25v12.6h6.3v6.2h6.2V180h-6.3v-6.3h-12.4v-6.2h-25v6.3h6.2v12.4h-6.3zm-143.8-6v-3.3h6.3v-25h-25v6.3H66.3v6.3H47.5V155h6.3v-6.3H41.3v6.3H35v-18.8h6.3V130h6.2v-6.3h6.3c9.8 0 18-1.6 31.4 6 5.7-.2 35.6.2 43.5.3v-6.3h12.6c13.5 6 12.8 9.5 18.7 18.8h-6.3v6.3h-6.2v6.2h-6.3v6.3H135v6.2h-6.3v6.3h-12.4v6.2H97.5zM60 145.5v-3h-6.3v6.2H60z" />
        <path id="shellShadow" fill="#785447" d="M141.3 108v-3h6.2V80h6.3v31.3h-6.3v6.4H141z" />
        <path id="clawLegsDarkStep1" fill="#9a4017" d="M235 164.4v-3.2h6.3V155h6.2v-6.3h6.3v18.8H235zm-37.5-15.7v-12.4h6.3v6.2h12.4V155H210v6.3h-12.5zM41.2 152v-3.2h6.3v-12.6h6.3V130H60v-12.5h-6.3v-6.3h12.5V105H85v6.3h6.3v6.2h6.2v6.3h6.3v6.2h12.5v6.3h-25v6.2H78.8v6.3H66.3v6.2h-25zm112.5-12.6v-3.2h-12.4v-18.6h25v18.7H160v6.2h-6.3z" />
        <path id="clawLegsDarkStep2" opacity="0" fill="#9a4017" d="M241.2 161.2V155h6.3v-6.3h6.2v18.8h-12.5zm-193.7-3V155h6.2v-6.3H60v-25h-6.3v-6.2h12.5v-6.3h12.5V105h12.5v12.5h6.3v6.2h6.2v6.3h12.5v6.2H110v6.3H97.5v6.2H78.7v6.3H66.2v6.2H47.5zm150-9.5v-12.5h6.2v6.3h12.5v18.7h-18.7zm-43.8-9.3v-3.2h-12.5v-18.7h25v18.7H160v6.3h-6.3z" />
        <path id="eyeNose" fill="#3f6533" d="M60 70.6v-3h-6.3v-6.3H60v6.2h31.3v6.3H60v-3.2zM41.2 58v-3h6.3v6.3h-6.3V58zM28.8 45.7v-3H35v6h-6.3v-3zm62.5-6.2v-3.2h6.3v6.3h-6.3v-3z" />
      </WalkingCrabigator>
    </SvgWrapper>
  </LayoutWrapper>
);

export default LoadingCrabigator;
