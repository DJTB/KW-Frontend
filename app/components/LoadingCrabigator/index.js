import React from 'react';
import styled, { keyframes } from 'styled-components';
import { transparent } from 'shared/styles/colors';

const fadeOut = keyframes`
  50% { opacity: 0; }
`;
const fadeIn = keyframes`
  50% { opacity: 1; }
`;

const LayoutWrapper = styled.div`
  display: flex;
  width: 100%;
  height: 100%;
  flex: 1 0 100%;
  justify-content: center;
  align-items: center;
`;

const SvgWrapper = styled.div`
  position: relative;
  width: 20vmin;
  height: 10vmin;
  background-repeat: no-repeat;
  background-color: ${transparent};
  transition: all 200ms ease-in-out;
`;

const WalkingCrabigator = styled.svg`
  display: block;
  pointer-events: none;
  transform-origin: 50% 50% 0px;
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;

  /* dance crabby dance */
  & path[id*="Step1"] {
    animation: ${fadeOut} 1s steps(1) infinite;
  }
  & path[id*="Step2"] {
    animation: ${fadeIn} 1s steps(1) infinite;
  }
`;

const LoadingCrabigator = () => (
  <LayoutWrapper>
    <SvgWrapper>
      <WalkingCrabigator
        title="Loading Animation"
        width="100%"
        height="100%"
        viewBox="0 0 400 200"
        preserveAspectRatio="xMinYMin meet"
      >
        <rect id="eyeBg" x="84" y="30" width="20px" height="19px" fill="#fff" />
        <rect id="neckBg" x="79" y="92" width="44px" height="38px" fill="#d0efdb" />
        <path id="pinkClawStep1" fill="#f36353" d="M222.5 183.12v-9.37h-6.25v-12.5H197.5v6.25h-6.25v6.25h-12.5V180h-6.25v6.25h-6.25V167.5h6.25v-6.25H160V155h-25v6.25h-6.25v6.25h-12.5v6.25h-12.5V180H85v-6.25H72.5v-12.5h6.25V155H85v-6.25h6.25v-6.25h-12.5v6.25h-12.5V155h-25v-6.25H35V130h6.25v-6.25h6.25v-6.25h6.25v-6.25h12.5V105H85v6.25h6.25v6.25h6.25v6.25h6.25V130h18.75V92.5h-6.25v6.25h-37.5V92.5H72.5v-6.25H60V80h-6.25v-6.25h-12.5V67.5H35v-6.25h-6.25V55H22.5V42.5h6.25v-6.25H47.5v6.25h6.25v-6.25h12.5V30h12.5v-6.25H97.5V30H110v6.25h12.5v6.25h6.25v6.25H135V55h6.25v12.5h6.25V80H160v-6.25h25V67.5h50v6.25h18.75V80h12.5v6.25h6.25v6.25h6.25v6.25H285v12.5h6.25v12.5h6.25v12.5H310v6.25h43.75v-6.25h12.5V130h6.25v12.5h-6.25v6.25H360V155h-18.75v6.25H272.5v6.25h6.25V180h-6.25v6.25h-6.25v-12.5H260v-6.25h-25v18.75h-6.25v6.25h-6.25v-9.38zm-131.25-12.5v-3.12H85v6.25h6.25v-3.13zm-43.75-25v-3.12h-6.25v6.25h6.25v-3.13zm50-6.25v-3.12h-6.25v6.25h6.25v-3.13zm6.25-96.87v-6.25H97.5V30h-6.25v6.25H85v6.25h6.25v6.25h12.5V42.5zm-12.5-3.13v-3.12h6.25v6.25h-6.25v-3.13z" />
        <path id="pinkClawStep2" opacity="0" d="M103.75 167.5h-12.5v6.25h-12.5v-12.5H85V155h6.25c0-7.286-.076-16.863-.076-16.863l12.966.675z" fill="#f36353" />
        <path id="shell" d="M228.75 167.5h-12.5v-6.25h-25c-9.06.021-12.288-.071-25 0H160V155h-12.883c.285-45.725.922-48.983.383-75H160v-6.25h25V67.5h50v6.25h18.75V80h12.5v6.25h6.25v6.25h6.25v6.25H285v12.5h6.25v12.5h6.25s-.426 5.22 0 12.5c1.606 27.452 4.465 22.973-18.75 25-8.497.742-12.5 20.64-12.5 6.25h-25z" fill="#ba8a59" />
        <path id="headTail" d="M122.5 92.5h-6.25v6.25h-37.5V92.5H72.5v-6.25H60V80h-6.25v-6.25h-12.5V67.5H35v-6.25h-6.25V55H22.5V42.5h6.25v-6.25H47.5v6.25h6.25v-6.25h12.5V30h12.5v-6.25H97.5V30H110v6.25h12.5v6.25h6.25v6.25H135V55h6.25v12.5h6.25V80h6.25c0 20.096-8.22 29.973-11.286 43.877-3.744 16.982-20.436 17.85-20.436 5.752 0-4.147.472-32.633.472-37.129zm-18.75-50v-6.25H97.5V30h-6.25v6.25H85v6.25h6.25v6.25h12.5zm-12.5-3.13v-3.12h6.25v6.25h-6.25zM285 155v-6.25h6.25v-12.5H310v6.25h43.75v-6.25h12.5V130h6.25v12.5h-6.25v6.25H360V155h-18.75v6.25H285z" fill="#8ac249" />
        <path id="clawLegsStep1" fill="#d45720" d="M222.5 183.12v-9.37h-6.25v-12.5H197.5v6.25h-6.25v6.25h-12.5V180h-6.25v6.25h-6.25V167.5h6.25v-6.25h6.25V130h18.75v6.25h6.25v6.25h18.75v6.25H235v12.5h6.25V155h6.25v-6.25h25v18.75h6.25V180h-6.25v6.25h-6.25v-12.5H260v-6.25h-25v18.75h-6.25v6.25h-6.25v-9.38zM85 176.87v-3.12h6.25v-12.5h6.25v-25h-6.25v6.25h-12.5v6.25h-12.5V155h-25v-6.25H35V130h6.25v-6.25h6.25v-6.25h6.25v-6.25h12.5V105H85v6.25h6.25v6.25h6.25v6.25h6.25V130h25v-6.25h12.5v-6.25h25v18.75H160v6.25h-6.25v6.25h-6.25V155H135v6.25h-6.25v6.25h-12.5v6.25h-12.5V180H85v-3.13zm-37.5-31.25v-3.12h-6.25v6.25h6.25v-3.13zm93.75-37.5V105h6.25V80h6.25v31.25h-12.5v-3.13zM60 70.62V67.5h-6.25v-6.25H60v6.25h31.25v6.25H60v-3.13zm-18.75-12.5V55h6.25v6.25h-6.25v-3.13zm-12.5-12.5V42.5H35v6.25h-6.25v-3.13zm62.5-6.25v-3.12h6.25v6.25h-6.25v-3.13z" />
        <path id="clawLegsStep2" opacity="0" d="M241.25 183.13V180h-12.5v-12.5h-12.5v-6.25h-25v6.25H185v6.25h-12.5V180h-18.75v-6.25H160v-6.25h6.25v-6.25h6.25V142.5h6.25V130h18.75v6.25h6.25v6.25h18.75v6.25H235v12.5h6.25V155h6.25v-6.25h25v12.5h6.25v6.25H285V180h-6.25v-6.25h-12.5v-6.25h-25v6.25h6.25v12.5h-6.25zM97.5 176.88v-3.13h6.25v-25h-25V155h-12.5v6.25H47.5V155h6.25v-6.25h-12.5V155H35v-18.75h6.25V130h6.25v-6.25h6.25c9.864 0 17.978-1.674 31.41 6.081 5.768-.333 35.667.114 43.59.169v-6.25h12.5c13.547 6.035 12.847 9.415 18.75 18.75h-6.25v6.25h-6.25V155h-6.25v6.25H135v6.25h-6.25v6.25h-12.5V180H97.5zM60 145.63v-3.13h-6.25v6.25H60z" fill="#d45720" />
        <path id="shellShadow" d="M141.25 108.1V105h6.25V80h6.25v31.26h-6.25l-.06 6.4-6.2-.06z" fill="#785447" />
        <path id="clawLegsDarkStep1" fill="#9a4017" d="M235 164.37v-3.12h6.25V155h6.25v-6.25h6.25v18.75H235zm-37.5-15.62v-12.5h6.25v6.25h12.5V155H210v6.25h-12.5zm-156.25 3.12v-3.12h6.25v-12.5h6.25V130H60v-12.5h-6.25v-6.25h12.5V105H85v6.25h6.25v6.25h6.25v6.25h6.25V130h12.5v6.25h-25v6.25h-12.5v6.25h-12.5V155h-25zm112.5-12.5v-3.12h-12.5v-18.7l6.25-.05h18.75v18.75H160v6.25h-6.25zM60 70.62V67.5h-6.25v-6.25H60v6.25h31.25v6.25H60zm-18.75-12.5V55h6.25v6.25h-6.25zm-12.5-12.5V42.5H35v6.25h-6.25zm62.5-6.25v-3.12h6.25v6.25h-6.25z" />
        <path id="clawLegsDarkStep2" opacity="0" d="M241.23 161.24V155h6.25v-6.26h6.26v18.75h-12.5zm-193.75-3.12V155h6.26v-6.26H60v-25h-6.26v-6.25h12.5v-6.28h12.5V105h12.5v12.5h6.25v6.23h6.24V130h12.5v6.24H110v6.25H97.5v6.22H78.74V155h-12.5v6.24H47.48zm150-9.38v-12.5h6.26v6.25h12.5v18.72H197.5zm-43.75-9.37v-3.13h-12.5v-12.5l.02-6.2 6.23-.06h18.75v18.74H160v6.25h-6.25z" fill="#9a4017" />
        <path id="eyeNose" d="M60 70.63V67.5h-6.25v-6.25H60v6.25h31.25v6.25H60v-3.13zm-18.75-12.5V55h6.25v6.25h-6.25v-3.13zm-12.5-12.5V42.5H35v6.25h-6.25v-3.13zm62.5-6.26v-3.12h6.25v6.25h-6.25v-3.13z" fill="#3f6533" />
        <path id="mouth" d="M28.75 45.63V42.5H35v6.25h-6.25v-3.13zm62.5-6.26v-3.12h6.25v6.25h-6.25v-3.13z" fill="#2f3e45" />
      </WalkingCrabigator>
    </SvgWrapper>
  </LayoutWrapper>
);

export default LoadingCrabigator;
