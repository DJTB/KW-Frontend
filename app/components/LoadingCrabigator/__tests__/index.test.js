import "jest-styled-components";
import { render } from "enzyme";
import React from "react";
import LoadingCrabigator from "../index";

describe("<LoadingCrabigator />", () => {
  it("should match snapshot", () => {
    const renderedComponent = render(<LoadingCrabigator />);
    expect(renderedComponent).toMatchSnapshot();
  });
});
