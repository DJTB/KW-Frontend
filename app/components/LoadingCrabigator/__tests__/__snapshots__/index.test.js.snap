// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadingCrabigator /> should match snapshot 1`] = `
.c0 {
  padding: 0.7875rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  height: 100%;
  -webkit-flex: 1 0 100%;
  -ms-flex: 1 0 100%;
  flex: 1 0 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
}

.c1 {
  position: relative;
  width: 30vmin;
  height: 20vmin;
  background-repeat: no-repeat;
  background-color: rgba(0,0,0,0);
  -webkit-transition: all 200ms ease-in-out;
  transition: all 200ms ease-in-out;
}

.c2 {
  display: block;
  pointer-events: none;
  -webkit-transform-origin: 50% 50% 0px;
  -ms-transform-origin: 50% 50% 0px;
  transform-origin: 50% 50% 0px;
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
}

.c2 path[id*="Step1"] {
  -webkit-animation: RabCa 1s steps(1) infinite;
  animation: RabCa 1s steps(1) infinite;
}

.c2 path[id*="Step2"] {
  -webkit-animation: doxdYu 1s steps(1) infinite;
  animation: doxdYu 1s steps(1) infinite;
}

@media (min-width:600px) {
  .c0 {
    padding: 1.333rem;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <svg
      class="c2"
      height="100%"
      preserveAspectRatio="xMinYMin meet"
      title="Loading Animation"
      viewBox="0 0 400 200"
      width="100%"
    >
      <path
        d="M84 30h20v19H84z"
        fill="#fff"
        id="eyeBg"
      />
      <path
        d="M135 155c1.3 1.7 0 6.3 0 6.3h-6.3v6.2h-12.4v6.3h-12.5v6.2H85v-6.3H72.5v-12.4h6.3V155H85v-6.3h6.3v-6.2S126 143 135 155zm-43.8 15.6v-3H85v6.2h6.3zm0-28h6.3z"
        fill="#f36353"
        id="pinkClawStep1"
      />
      <path
        d="M79 92h44v38H79z"
        fill="#d0efdb"
        id="neckBg"
      />
      <path
        d="M103.8 167.5H91.3v6.3H78.8v-12.6H85V155h6.3v-17l12.8.8z"
        fill="#f36353"
        id="pinkClawStep2"
        opacity="0"
      />
      <path
        d="M228.8 167.5h-12.6v-6.3H160V155h-13c.4-45.7 1-49 .5-75H160v-6.3h25v-6.2h50v6.3h18.8V80H266v6.3h6.3v6.2h6.3v6.3h6.2v12.5h6.3v12.5h6.6s-.4 5.2 0 12.5c1.6 27.4 4.5 24.5-18.8 25-20.7.3-12-3.8-12.4 6.2h-25z"
        fill="#ba8a59"
        id="shell"
      />
      <path
        d="M122.5 92.5h-6.3v6.3H78.8v-6.3h-6.2v-6.3H60V80h-6.3v-6.3H41.3v-6.2H35v-6.3h-6.3V55h-6.2V42.5h6.3v-6.3h18.7v6.3h6.3v-6.3h12.5V30h12.5v-6.3h18.7V30H110v6.3h12.5v6.2h6.3v6.3h6.2V55h6.3v12.5h6.2V80h6.3c0 20-8.3 30-11.3 44-3.8 17-20.5 17.7-20.5 5.6l.5-37zm-18.8-50v-6.3h-6.2V30h-6.3v6.3H85v6.2h6.3v6.3h12.5zm-12.5-3V36h6.3v6.3h-6.3zM285 155v-6.3h6.3v-12.4H310v6.2h43.8v-6.3H366V130h6.3v12.5H366v6.3h-6v6.2h-18.8v6.3H285z"
        fill="#8ac249"
        id="headTail"
      />
      <path
        d="M222.5 183v-9h-6.3v-12.6h-18.7v6.3h-6.3v6.3h-12.4v6.2h-6.3v6.3h-6.3v-18.8h6.3v-6.3h6.3V130h18.7v6.3h6.3v6.2h18.7v6.3H235v12.4h6.3V155h6.2v-6.3h25v18.8h6.3V180h-6.3v6.3h-6.3v-12.6H260v-6.2h-25v18.8h-6.3v6.2h-6.2zM85 177v-3.2h6.3v-12.6h6.2v-25h-6.3v6.3H78.8v6.3H66.3v6.2h-25v-6.3H35V130h6.3v-6.3h6.2v-6.2h6.3v-6.3h12.5V105H85v6.3h6.3v6.2h6.2v6.3h6.3v6.2h25v-6.3h12.4v-6.2h25v18.8H160v6.2h-6.3v6.3h-6.2v6.2H135v6.3h-6.3v6.2h-12.4v6.3h-12.5v6.2H85zm-37.5-31.4v-3h-6.3v6.2h6.3z"
        fill="#d45720"
        id="clawLegsStep1"
      />
      <path
        d="M241.3 183v-3h-12.6v-12.5h-12.4v-6.3h-25v6.3H185v6.3h-12.5v6.2h-18.8v-6.3h6.3v-6.2h6.3v-6.3h6.2v-18.7h6.3V130h18.7v6.3h6.3v6.2h18.7v6.3H235v12.4h6.3V155h6.2v-6.3h25v12.6h6.3v6.2h6.2V180h-6.3v-6.3h-12.4v-6.2h-25v6.3h6.2v12.4h-6.3zm-143.8-6v-3.3h6.3v-25h-25v6.3H66.3v6.3H47.5V155h6.3v-6.3H41.3v6.3H35v-18.8h6.3V130h6.2v-6.3h6.3c9.8 0 18-1.6 31.4 6 5.7-.2 35.6.2 43.5.3v-6.3h12.6c13.5 6 12.8 9.5 18.7 18.8h-6.3v6.3h-6.2v6.2h-6.3v6.3H135v6.2h-6.3v6.3h-12.4v6.2H97.5zM60 145.5v-3h-6.3v6.2H60z"
        fill="#d45720"
        id="clawLegsStep2"
        opacity="0"
      />
      <path
        d="M141.3 108v-3h6.2V80h6.3v31.3h-6.3v6.4H141z"
        fill="#785447"
        id="shellShadow"
      />
      <path
        d="M235 164.4v-3.2h6.3V155h6.2v-6.3h6.3v18.8H235zm-37.5-15.7v-12.4h6.3v6.2h12.4V155H210v6.3h-12.5zM41.2 152v-3.2h6.3v-12.6h6.3V130H60v-12.5h-6.3v-6.3h12.5V105H85v6.3h6.3v6.2h6.2v6.3h6.3v6.2h12.5v6.3h-25v6.2H78.8v6.3H66.3v6.2h-25zm112.5-12.6v-3.2h-12.4v-18.6h25v18.7H160v6.2h-6.3z"
        fill="#9a4017"
        id="clawLegsDarkStep1"
      />
      <path
        d="M241.2 161.2V155h6.3v-6.3h6.2v18.8h-12.5zm-193.7-3V155h6.2v-6.3H60v-25h-6.3v-6.2h12.5v-6.3h12.5V105h12.5v12.5h6.3v6.2h6.2v6.3h12.5v6.2H110v6.3H97.5v6.2H78.7v6.3H66.2v6.2H47.5zm150-9.5v-12.5h6.2v6.3h12.5v18.7h-18.7zm-43.8-9.3v-3.2h-12.5v-18.7h25v18.7H160v6.3h-6.3z"
        fill="#9a4017"
        id="clawLegsDarkStep2"
        opacity="0"
      />
      <path
        d="M60 70.6v-3h-6.3v-6.3H60v6.2h31.3v6.3H60v-3.2zM41.2 58v-3h6.3v6.3h-6.3V58zM28.8 45.7v-3H35v6h-6.3v-3zm62.5-6.2v-3.2h6.3v6.3h-6.3v-3z"
        fill="#3f6533"
        id="eyeNose"
      />
    </svg>
  </div>
</div>
`;
